common/workunit/workunit.cpp

```diff
@@ -6359,6 +6429,83 @@ unsigned CLocalWorkUnit::queryFileUsage(const char *fileName) const
     return p->getPropInt(path.str());
 }
 
+IConstWUFileUsageIterator * CLocalWorkUnit::getFieldUsage() const
+{
+    CriticalBlock block(crit);
+    IPropertyTree* fieldUsageTree = p->queryPropTree("usedsources");
+
+    if (!fieldUsageTree)
+        return NULL;
+
+    IPropertyTreeIterator* iter = fieldUsageTree->getElements("*");
+    return new CConstWUFileUsageIterator(iter);
+}
+
+bool CLocalWorkUnit::getFieldUsageArray(StringArray & filenames, StringArray & columnnames, const char * clusterName) const
+{
+    bool scopeLoaded = false;
+    SCMStringBuffer defaultScope;
+
+    Owned<IConstWUFileUsageIterator> files = getFieldUsage();
+
+    if (!files)
+        return false; // this query was not compiled with recordFieldUsage option.
+
+    ForEach(*files)
+    {    
+        Owned<IConstWUFileUsage> file = files->get();
+
+        SCMStringBuffer filename;
+        file->getName(filename);
+
+        // filename cannot be empty
+        if (filename.length() < 3)
+            throw MakeStringException(WUERR_InvalidFieldUsage, "Invalid FieldUsage found in WU. Cannot enforce view security.");
+
+        // Filenames in field usage are surrounded with single quotes ('filename').
+        // Need to remove surrounding single quotes quickly using memcpy.
+        char* cleanFilename = new char[filename.length()-1];
+        memcpy(cleanFilename, filename.str()+1, filename.length()-2);
+        cleanFilename[filename.length()-2] = '\0';
+        filename.set(cleanFilename); 
```

Comment:
easier to say

StringAttr cleanFilename(cleanFilename.str()+1, cleanFilename.length()-2);